src/Base/Identifier.php

```diff
@@ -19,23 +19,42 @@
  */
 class Identifier extends ValueObject
 {
-    public function __construct(UuidInterface $identifier)
-    {
-        parent::__construct($identifier);
-    }
+    /**
+     * @var UuidInterface
+     */
+    private $value;
 
-    public function equals(ValueObject $identifier): bool
+    /**
+     * Identifier constructor.
+     * @param UuidInterface $identifier
+     */
+    public function __construct(UuidInterface $identifier)
     {
-        return (string) $this == (string) $identifier;
+        $this->value = $identifier;
     }
 
+    /**
+     * @return Identifier
+     */
     public static function generate(): self
     {
         return new static(Uuid::uuid1());
     }
 
+    /**
+     * @param string $identifier
+     * @return Identifier
+     */
     public static function fromString(string $identifier): self
     {
         return new static(Uuid::fromString($identifier));
     }
+
+    /**
+     * @return string
+     */
+    public function __toString(): string
```

Comment:
Just a question, isn't it better to keep the __toString in the ValueObject? Normally we are casting it to string, so if a class by definition needs something more complex we can implement it.

I see what you wanted to do, to provide a property that is called as the value it represents. This was something I was thinking and I thought it could be a trade off to avoid duplication of code (toString).